packages/core/src/crypto/Jwk.ts

```diff
@@ -0,0 +1,84 @@
+import type { Ed25519Jwk, Jwk, P256Jwk, P384Jwk, P521Jwk, X25519Jwk } from './JwkTypes'
+import type { Key } from './Key'
+
+import { TypedArrayEncoder, Buffer } from '../utils'
+
+import { compress, expand } from './EcCompression'
+import {
+  isBls12381g1Jwk,
+  isBls12381g2Jwk,
+  jwkCurveToKeyTypeMapping,
+  keyTypeToJwkCurveMapping,
+  isEd25519Jwk,
+  isX25519Jwk,
+  isP256Jwk,
+  isP384Jwk,
+  isP521Jwk,
+} from './JwkTypes'
+import { KeyType } from './KeyType'
+
+export function getKeyDataFromJwk(jwk: Jwk): { keyType: KeyType; publicKey: Uint8Array } {
+  // ed25519, x25519
+  if (isEd25519Jwk(jwk) || isX25519Jwk(jwk)) {
+    return {
+      publicKey: TypedArrayEncoder.fromBase64(jwk.x),
+      keyType: jwkCurveToKeyTypeMapping[jwk.crv],
+    }
+  }
+
+  // bls12381g1, bls12381g2
+  if (isBls12381g1Jwk(jwk) || isBls12381g2Jwk(jwk)) {
+    return {
+      publicKey: TypedArrayEncoder.fromBase64(jwk.x),
+      keyType: jwkCurveToKeyTypeMapping[jwk.crv],
+    }
+  }
+
+  // p-256, p-384, p-521
+  if (isP256Jwk(jwk) || isP384Jwk(jwk) || isP521Jwk(jwk)) {
+    // TODO: do we want to use the compressed key in the Key instance?
+    const publicKeyBuffer = Buffer.concat([TypedArrayEncoder.fromBase64(jwk.x), TypedArrayEncoder.fromBase64(jwk.y)])
```

Comment:
I think the did:key spec only supports the compressed variant (that's what I understand from the thread), but you're saying the JWK itself could contain both compressed and uncompressed variants? I think we could just support uncompressed as well (as it's just skipping the compression/uncompression and thus simpler). 

We'll always encode to compressed variant (because why not), but for decompression I should just look if the key is prefixed with `0x3` or `0x2` and if so, remove that previx and not do the decompression?

Reply:
AFAIK the JWK spec does not allow for compressed values.

Reply:
Hmm in that case, there's no reason to support the uncompressed variant I think? did:key only supports compressed, and the jwk will have separate values (x and y), not a single concatenated public key

Reply:
Agreed. We can always add uncompressed later on, if needed.

Reply:
> AFAIK the JWK spec does not allow for compressed values.

The JWK spec does very much support compressed and uncompressed. We are talking specifically about key representation and not the compressed format of a jwt, etc.

Reply:
> I think the did:key spec only supports the compressed variant (that's what I understand from the thread), but you're saying the JWK itself could contain both compressed and uncompressed variants? 

No, I mean that we should check whether it is compressed or uncompressed, based on the prefix, and handle accordingly. We will break keys if this mechanism is implemented incorrectly which will lead to a lot of errors. 

> I think we could just support uncompressed as well (as it's just skipping the compression/uncompression and thus simpler). 
 
Yes! I don't think it's too much effort. 


> We'll always encode to compressed variant (because why not), but for decompression I should just look if the key is prefixed with `0x3` or `0x2` and if so, remove that previx and not do the decompression?

Yes, I would have to look at the exact algo for this, but I think that's it.

Reply:
@blu3beri I've addressed all comments except the ones related to compression of P keys. Can you help me out a bit in pointing out what I need to exactly?